response_operations_ui/views/accounts.py

```diff
@@ -11,33 +11,83 @@
-            "email": user_from_uaa["emails"][0]["value"],
-            "password_last_changed": formatted_date,
-        }
-        return render_template("account/my-account.html", user=user)
-    except Exception as e:
-        return jsonify(e)
+    form = MyAccountOptionsForm()
+    form_valid = form.validate()
+    if form_valid and request.method == "POST":
+        return redirect(url_for(form_redirect_mapper.get(form.data["option"])))
+    elif not form_valid and request.method == "POST":
+        flash("You need to choose an option")
+        return redirect(url_for("account_bp.get_my_account"))
+    user_id = session["user_id"]
+    user_from_uaa = uaa_controller.get_user_by_id(user_id)
+    first_name = user_from_uaa["name"]["givenName"]
+    last_name = user_from_uaa["name"]["familyName"]
+    formatted_date = dates.format_datetime_to_string(user_from_uaa["passwordLastModified"], date_format="%d %b %Y")
+    user = {
+        "username": user_from_uaa["userName"],
+        "name": f"{first_name} {last_name}",
+        "email": user_from_uaa["emails"][0]["value"],
+        "password_last_changed": formatted_date,
+    }
+    return render_template("account/my-account.html", user=user)
+
+
+@account_bp.route("/change-account-name", methods=["GET", "POST"])
+@login_required
+def change_account_name():
+    form = ChangeAccountName()
+    user_id = session["user_id"]
+    user_from_uaa = uaa_controller.get_user_by_id(user_id)
+    user = {
+        "first_name": f"{user_from_uaa['name']['givenName']}",
+        "last_name": f"{user_from_uaa['name']['familyName']}",
+    }
+    user_from_uaa["name"] = {"familyName": form.data["last_name"], "givenName": form.data["first_name"]}
+    if request.method == "POST" and form.validate():
+        if (form.data["first_name"] != user["first_name"]) or (form.data["last_name"] != user["last_name"]):
+            errors = uaa_controller.update_user_account(user_from_uaa)
+            if errors is None:
+                full_name = f"{form.data['first_name']} {form.data['last_name']}"
+                logger.info("Sending update account details email", user_id=user_id)
+                personalisation = {
+                    "first_name": user["first_name"],
+                    "value_name": "name",
+                    "changed_value": full_name,
+                }
+                try:
+                    NotifyController().request_to_notify(
+                        email=user_from_uaa["emails"][0]["value"],
+                        template_name="update_account_details",
+                        personalisation=personalisation,
+                    )
+                    return redirect(url_for("logout_bp.logout", message="Your name has been changed"))
+                except NotifyError as e:
+                    logger.error(
+                        "Error sending change of name acknowledgement email to Notify Gateway", msg=e.description
+                    )
```

Comment:
You'll want to actually do something if theres a notifyError that isn't just logging something out.  I don't think it has to be much as the update has already happened in uaa.  Maybe `flash("Something went wrong when sending an email to confirm the change.  Your details have been updated in the system")